### ISSUE TITLE
Improved inferred type for Records

### ISSUE THREAD

**Issue by Thanaen** [fabian-hiller/valibot#10]
Take this object, created with zod:
```ts
const ZodProblemDetailsSchema = z.object({
  errors: z.record(z.array(z.string())).optional(),
});
```

When we infer the type of this schema with the z.infer method, we see that the type inferred for the Record is quite simple:
```ts
type ZodProblemDetails = {
    errors?: Record<string, string[]> | undefined;
}
```

I've implemented the same thing with valibot:
```ts
const ProblemDetailsSchema = object({
  errors: optional(record(array(string()))),
});
```

The type inferred with Output gives this: 
```ts
type ProblemDetails = {
    errors: RecordOutput<StringSchema, ArraySchema<StringSchema<string>, string[]>> | undefined;
}
```

Would it be possible to come up with a type similar to zod Records for valibot Records? They seem more readable / simple to understand!

Thank you!

**Comment by fabian-hiller** [comment]
Thanks for the hint. I will have a look at it. Maybe I made a mistake or it's TypeScript. Is the type, apart from what TypeScript displays, correct?

**Comment by Thanaen** [comment]
According to my tests, the type generated accepts the same data as the one generated by Zod, so I deduce that it is correct / functional!

**Comment by fabian-hiller** [comment]
Thanks for the info. Then it seems to be only a visual TypeScript "problem".

**Comment by fabian-hiller** [comment]
I'll still check to see if it can be controlled.

**Comment by fabian-hiller** [comment]
Should be fixed in v0.8.0.
